src/coordination/coordinator_instance.cpp

```diff
@@ -352,24 +315,57 @@ auto CoordinatorInstance::ShowInstances() const -> std::vector<InstanceStatus> {
   return std::move(maybe_res.value());
 }
 
-auto CoordinatorInstance::ForceResetCluster_() -> ForceResetClusterStateStatus {
-  // Force reset tries to return cluster to state in which we have all the replicas we had before
-  // and try to do failover to new MAIN. Only then is force reset successful
+auto CoordinatorInstance::VerifyOrCorrectClusterState() -> VerifyOrCorrectClusterStateStatus {
+  utils::ExponentialBackoff backoff{std::chrono::milliseconds(1000), std::chrono::seconds(60)};
 
-  // 0. Open lock
-  // 1. Try to demote each instance to replica
-  // 2. Instances which are demoted proceed in next step as part of selection process
+  while (raft_state_->IsLeader()) {
+    spdlog::trace("Ensuring healthy state in cluster still as coordinator is leader and lock is opened.");
+    auto const result = VerifyOrCorrectClusterState_();
+    switch (result) {
+      case (VerifyOrCorrectClusterStateStatus::SUCCESS):
+        is_leader_ready_.store(true, std::memory_order_release);
```

Comment:
here we are using memory_order_release while in GetBecomeLeaderCallback we are using seq_cst. I vote for release and acquire, should be enough and is consistent. Do you have any specific reason why you used seq_cst?

Reply:
`seq_cst`